File: math/armstrong_number/Java/Armstrong.java

/* package whatever; // don't place package name! */

import java.util.*;
import java.lang.*;
import java.io.*;
import java.math.BigInteger;
/* Name of the class has to be "Main" only if the class is public. */
class Armstrong
{
	public static boolean isArmstrong(String num){

		BigInteger sumVal = BigInteger.valueOf(0);
		int numOfDigits = num.length();
		for(int i=0;i<numOfDigits;i++){
			String dig = "";
			dig += num.charAt(i);
			BigInteger digit = new BigInteger(dig);
			sumVal = sumVal.add(digit.pow(numDigits));
		}
		if(sumVal.toString().equals(num)){
			return true;
		}
		return false;
	}
	public static void main (String[] args) throws java.lang.Exception
	{
		// your code goes here
		// System.out.println(isArmstrong("1634"));
		if(isArmstrong("1634")){
			System.out.println("Number is Armstrong");
		}
		else{
			System.out.println("Number is not Armstrong");	
		}
	}
}

File: math/armstrong_number/C/armstrong.c
#include<stdio.h>
#include<math.h>

int main(void)
{
	int n,r,a=0,t,c=0,z;
	printf("Enter a number: ");
	scanf("%d",&n);
	t=n; z=n;
	while(z>0)
    {
       z=z/10;
       c++;
    }
	while(n>0)
    {
        r=n%10;
        a=a+pow(r,c);
        n=n/10;
    }
    if(a==t)
        printf("\nThe number is armstrong\n");
    else printf("\nThe number is not armstrong\n");
    return 0;
}
